modules/standard/Socket.chpl

```diff
@@ -0,0 +1,1282 @@
+/*
+ * Copyright 2020-2021 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+  Socket Module library, specifically related to IP Sockets.
+
+  The Socket module focuses on connecting, accepting sockets and providing interface for
+  communication between the sockets. Also provided are some constant values representing
+  common idioms in socket programming, such as standard Addresses, Families and Flags.
+  For more constants that can used alongside refer :mod:`Sys`.
+
+  To those familiar with the Unix socket API, the method names will feel familiar,
+  though their usage will be somewhat simpler than the raw Unix socket API.
+
+  Records
+  -----------------
+  :type:`ipAddr`
+  :type:`tcpConn`
+  :type:`tcpListener`
+  :type:`udpSocket`
+
+  Enum
+  -----------------
+  :type:`IPFamily`
+
+  Procedures
+  -----------------
+  :proc:`bind`
+  :proc:`connect`
+  :proc:`delayAck`
+  :proc:`getpeername`
+  :proc:`getSockOpt`
+  :proc:`getsockname`
+  :proc:`listen`
+  :proc:`naggle`
+  :proc:`setSockOpt`
+
+  Records, Types and Function Definitions
+  ---------------------------------------
+*/
+module Socket {
+
+// TODO: replace with mason config
```

Comment:
Somewhere, we need a description of what the implementation is doing with libevent at a high level. It could be here in a comment (that is not part of the docs) or it could be in the PR message itself.

Reply:
I am not sure what you mean by implementation

Reply:
Like, can you make a comment that is a sort of road map / outline for this `Socket.chpl` file? I'm not so worried about outlining the different sections of implementation. I'd like it to describe how the Socket module is using libevent, at a high level. (The low level description can be the source code, but a high level view can be hard for someone reading the source code to get).